plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/rest/TrinoIcebergRestCatalogFactory.java

```diff
@@ -81,13 +83,18 @@ public synchronized TrinoCatalog create(ConnectorIdentity identity)
             warehouse.ifPresent(location -> properties.put(CatalogProperties.WAREHOUSE_LOCATION, location));
             properties.put("trino-version", trinoVersion);
             properties.putAll(securityProperties.get());
+
+            if (vendedCredentialsEnabled) {
+                properties.put("header.x-tabular-s3-access", "vended_credentials");
+            }
+
             RESTSessionCatalog icebergCatalogInstance = new RESTSessionCatalog(
                     config -> HTTPClient.builder(config).uri(config.get(CatalogProperties.URI)).build(),
                     (context, config) -> {
                         ConnectorIdentity currentIdentity = (context.wrappedIdentity() != null)
                                 ? ((ConnectorIdentity) context.wrappedIdentity())
                                 : ConnectorIdentity.ofUser("fake");
-                        return new ForwardingFileIo(fileSystemFactory.create(currentIdentity));
+                        return new ForwardingFileIo(fileSystemFactory.create(currentIdentity, config), config);
```

Comment:
Why do we need to provide the properties to `ForwardingFileIo`?

Reply:
The other spots that get the vended credentials via the FileIO like `transaction.table().io().properties()` all use them.